### PR TITLE
[videodb] Movie Deletion/Conversion to Version Refactor and Optimization

### PR DESCRIPTION
**Pull request by CrystalP**
## Description

Minor refactor and optimization in movie deletion and conversion of a movie into a version.

See motivation and context for the "why".

One commit per logical step, as follows:

[refactors]
- Removed the KeepId parameter of DeleteMovie. It was complicating the flow and reducing the function to a stream deletion. Have the only caller delete the streams directly instead.
- The stream deletion by the caller is redundant because of an unconditional call a bit later of "SetStreamDetailsForFileId" which unconditionally deletes all streams first, before adding the new ones (if any) > removed the duplicate stream deletion
- Added doxygen to SetStreamDetailsForFileId

[small optimization]
- Added parameter to DeleteMovie to skip the hash deletion for the case of movie conversion into a version. That doesn't disrupt the flow of the function and is OK I think

No backport to v21, there is nothing broken to fix.

## Motivation and context

Prompted by an investigation of what happens after a movie is turned into a version, and the user refreshes the library or adds the movie to library again from Videos node.

There is no crash and things work fine, but a small optimization can be made to the hash: it's deleted when converting a movie into a version, but recreated (even though there is no file/folder change) in the next library scan / explicit scan of the movie.

Instead, the hash could be preserved and the next scan will be a little faster (detect the folder as "no change" instead of "new")

Then noticed that the parameters of the DeleteMovie are a bit strange and could use a bit of refactoring.

## How has this been tested?

Deleted movies, converted movies to versions
Tried different scraper settings (single movie in folder named after the movie or not, recurse or not)
"Library Update" function, "Scan to library", "Scan source"

## What is the effect on users?

Barely noticeable - slightly faster library scan after converting movies to versions

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [X] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **Student submission** (PR was done for educational purposes and will be treated as such)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by CrystalP** [comment]
jenkins please build this

File: xbmc/video/VideoDatabase.cpp [hunk 1]
```diff
@@ -3716,55 +3712,49 @@ void CVideoDatabase::DeleteMovie(int idMovie,
...
-        // Clean up the other assets attached to the movie, if any.
+    if (ca == DeleteMovieCascadeAction::ALL_ASSETS)
+    {
+      // The default version of the movie was removed by a delete trigger.
+      // Clean up the other assets attached to the movie, if any.
 
-        // need local dataset due to nested DeleteVideoAsset query
-        std::unique_ptr<Dataset> pDS{m_pDB->CreateDataset()};
+      // need local dataset due to nested DeleteVideoAsset query
+      std::unique_ptr<Dataset> pDS{m_pDB->CreateDataset()};
```
Comment (ksooo) [inline xbmc/video/VideoDatabase.cpp:3736]: const std::unique_ptr … ?

**Review by ksooo — APPROVED** [review]
Not sure i understand the code change 100% but i send not spotted anything suspicious.

File: xbmc/video/VideoDatabase.cpp (hunk 1 above)
Comment (CrystalP) [reply xbmc/video/VideoDatabase.cpp]: yes. updated and rebased the PR.

**Pull request merged**